shared/src/router-config/graphql-configuration.ts

```diff
@@ -99,25 +101,54 @@ export function configurationDataMapToDataSourceConfiguration(
     addRequiredFields(data.keys, output.keys, typeName);
     addRequiredFields(data.provides, output.provides, typeName);
     addRequiredFields(data.requires, output.requires, typeName);
+    const natsEventConfigurations: NatsEventConfiguration[] = [];
+    const kafkaEventConfigurations: KafkaEventConfiguration[] = [];
     for (const event of data.events ?? []) {
-      output.events.push(
-        new EventConfiguration({
-          fieldName: event.fieldName,
-          sourceName: event.sourceName,
-          subjects: event.subjects,
-          type: eventType(event.type),
-          typeName,
-          ...(event.streamConfiguration
-            ? {
-                streamConfiguration: new StreamConfiguration({
-                  consumerName: event.streamConfiguration.consumerName,
-                  streamName: event.streamConfiguration.streamName,
-                }),
-              }
-            : {}),
-        }),
-      );
+      switch (event.providerType) {
+        case 'kafka': {
+          kafkaEventConfigurations.push(
+            new KafkaEventConfiguration({
+              engineEventConfiguration: new EngineEventConfiguration({
+                fieldName: event.fieldName,
+                providerId: event.providerId,
+                type: eventType(event.type),
+                typeName,
+              }),
+              topics: event.topics,
+            }),
+          );
+          break;
+        }
+        case 'nats': {
+          natsEventConfigurations.push(
+            new NatsEventConfiguration({
+              engineEventConfiguration: new EngineEventConfiguration({
+                fieldName: event.fieldName,
+                providerId: event.providerId,
+                type: eventType(event.type),
+                typeName,
+              }),
+              subjects: event.subjects,
+              ...(event.streamConfiguration
+                ? {
+                    streamConfiguration: new NatsStreamConfiguration({
+                      consumerName: event.streamConfiguration.consumerName,
+                      streamName: event.streamConfiguration.streamName,
+                    }),
+                  }
+                : {}),
+            }),
+          );
+          break;
+        }
+        default: {
+          // TODO propagate this properly
+          throw new Error(`Unknown event provider.`);
```

Comment:
How do we want to handle this? Theoretically, it should never happen.

Reply:
This is a question for you. Ideally, we can exhaust the switch by using types. I'm fine with both implementation. An error would highlight that something significantly is broken.